flow/state.go

```diff
@@ -0,0 +1,60 @@
+package flow
+
+import (
+	"gopkg.in/telebot.v3"
+)
+
+// StateHandler is a common handler for global processes, such as finally, fail, step, and so on.
+type StateHandler func(state *State) error
+
+// State defines the user's state and persists common elements, such as the bot instance, flow handler instance, etc.
+type State struct {
```

Comment:
This feels a little backwards, I would much rather have State implement some kind of interface